gson/src/main/java/com/google/gson/internal/bind/ReflectiveTypeAdapterFactory.java

```diff
@@ -94,22 +98,32 @@ private List<String> getFieldNames(Field f) {
     return fieldNames;
   }
 
-  @Override public <T> TypeAdapter<T> create(Gson gson, final TypeToken<T> type) {
+  @Override
+  public <T> TypeAdapter<T> create(Gson gson, final TypeToken<T> type) {
     Class<? super T> raw = type.getRawType();
 
     if (!Object.class.isAssignableFrom(raw)) {
       return null; // it's a primitive!
     }
 
-    FilterResult filterResult = ReflectionAccessFilterHelper.getFilterResult(reflectionFilters, raw);
+    if (ReflectionHelper.isRecord(raw)) {
+      // We can assume blockInaccessible here, because it will never kick inn. We are not doing
```

Comment:
I first thought this would only apply to writing fields, but I see the accessible flag is also needed to read values via BoundedField. The RecordComponent actually has an accessor field, that returns the getter method, and I think the most correct approach here would actually be to use the accessor method. Then we do not need to use the set accessible at all. Still, I guess we need to evaluate the reflection filtering result. I will push a fix to this.